### PR TITLE
Add KubeOne and extend Kubermatic entries in production environment tools

### PR DESCRIPTION
**Pull request by eqrx**
I would like to add [KubeOne](https://github.com/kubermatic/kubeone) as a possibility in https://kubernetes.io/docs/setup/ .

I also reviewed the Kubermatic entry and I am unsure about the current state of it. Do you mean by "custom" (last the columns) that you can configure or modify listed components? Kubermatic sets up the components as vanilla as possible but you can modify that behavior by using addons. 

Thanks!

### CODE REVIEW

**Comment by eqrx** [comment]
I signed it

**Comment by eqrx** [comment]
I signed it

File: content/en/docs/setup/_index.md [hunk 1]
```diff
@@ -88,7 +88,8 @@ The following production environment solutions table lists the providers and the
 | [IBM](https://www.ibm.com/in-en/cloud)        | [IBM Cloud Kubernetes Service](https://cloud.ibm.com/kubernetes/catalog/cluster)| |[IBM Cloud Private](https://www.ibm.com/in-en/cloud/private) | |
 | [Ionos](https://www.ionos.com/enterprise-cloud)        | [Ionos Managed Kubernetes](https://www.ionos.com/enterprise-cloud/managed-kubernetes) | [Ionos Enterprise Cloud](https://www.ionos.com/enterprise-cloud) | |
 | [Kontena Pharos](https://www.kontena.io/pharos/)          |    |&#x2714;|    &#x2714;    |        |         |
-| [Kubermatic](https://www.loodse.com/)         |     &#x2714;  | &#x2714; | &#x2714;     |     |  |
+| [Kubermatic](https://kubermatic.io/) | &#x2714; | &#x2714; | &#x2714; | &#x2714; | &#x2714; |  |
+| [Kubeone](https://kubeone.io/) |  | &#x2714; | &#x2714; | &#x2714; | &#x2714; | &#x2714; |
```
Comment (tengqm) [inline content/en/docs/setup/_index.md]: move this before line 91?

Comment (eqrx) [reply content/en/docs/setup/_index.md]: Done, thanks! Have also changed the link of KubeOne!

**Pull request merged**